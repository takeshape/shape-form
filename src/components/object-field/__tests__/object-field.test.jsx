import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ReactShallowRenderer();
import ObjectField from '../index';

describe('ObjectField', () => {
  it('renders without required', () => {
    const schema = {
      type: 'object',
      properties: {
        email: {type: 'string'},
        firstName: {type: 'string'},
        lastName: {type: 'string'}
      }
    };
    const props = {
      schema,
      formName: 'test-form',
      path: '',
      config: {},
      context: {foo: 'bar'},
      widgets: {},
      locale: 'pt-br'
    };
    const tree = renderer.render(<ObjectField {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders with required', () => {
    const schema = {
      type: 'object',
      properties: {
        email: {type: 'string'},
        firstName: {type: 'string'},
        lastName: {type: 'string'}
      },
      required: ['email']
    };
    const props = {
      schema,
      formName: 'test-form',
      path: 'user',
      config: {},
      locale: 'pt-br'
    };
    const tree = renderer.render(<ObjectField {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders with title and required', () => {
    const schema = {
      title: 'User',
      type: 'object',
      properties: {
        firstName: {type: 'string'},
        lastName: {type: 'string'},
        email: {type: 'string'}
      }
    };
    const props = {
      schema,
      formName: 'test-form',
      path: 'user',
      config: {},
      isRequired: true,
      locale: 'pt-br'
    };
    const tree = renderer.render(<ObjectField {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders with configured order', () => {
    const schema = {
      type: 'object',
      properties: {
        firstName: {type: 'string'},
        lastName: {type: 'string'},
        email: {type: 'string'}
      },
      required: ['email']
    };
    const props = {
      schema,
      formName: 'test-form',
      path: 'user',
      config: {
        order: ['email', 'firstName', 'lastName']
      },
      locale: 'pt-br'
    };
    const tree = renderer.render(<ObjectField {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders with empty order', () => {
    const schema = {
      type: 'object',
      properties: {
        firstName: {type: 'string'},
        lastName: {type: 'string'},
        email: {type: 'string'}
      },
      required: ['email']
    };
    const props = {
      schema,
      formName: 'test-form',
      path: 'user',
      config: {
        order: []
      },
      locale: 'pt-br',
      context: {
        projectId: 'project_id',
        contentTypeId: 'content_type_id',
        contentId: 'content_id'
      }
    };
    const tree = renderer.render(<ObjectField {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('handles invalid property names in order', () => {
    const schema = {
      type: 'object',
      properties: {
        firstName: {type: 'string'},
        lastName: {type: 'string'},
        email: {type: 'string'}
      },
      required: ['email']
    };
    const props = {
      schema,
      formName: 'test-form',
      path: 'user',
      config: {
        order: ['email', 'bogus', 'firstName', 'lastName']
      },
      locale: 'pt-br'
    };
    const tree = renderer.render(<ObjectField {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders emptyObject widget', () => {
    const schema = {
      type: 'object',
      properties: {}
    };
    const props = {
      schema,
      widgets: {
        emptyObject: 'EmptyObject'
      },
      formName: 'test-form',
      path: 'user',
      config: {},
      locale: 'pt-br',
      context: {}
    };
    const tree = renderer.render(<ObjectField {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
