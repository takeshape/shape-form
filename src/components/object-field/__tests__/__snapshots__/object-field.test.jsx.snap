// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectField handles invalid property names in order 1`] = `
<React.Fragment>
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={true}
    locale="pt-br"
    path="user.email"
    schema={
      Object {
        "type": "string",
      }
    }
  />
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="user.firstName"
    schema={
      Object {
        "type": "string",
      }
    }
  />
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="user.lastName"
    schema={
      Object {
        "type": "string",
      }
    }
  />
</React.Fragment>
`;

exports[`ObjectField renders emptyObject widget 1`] = `
<EmptyObject
  config={Object {}}
  context={Object {}}
  formName="test-form"
  locale="pt-br"
  path="user"
  schema={
    Object {
      "properties": Object {},
      "type": "object",
    }
  }
  widgets={
    Object {
      "emptyObject": "EmptyObject",
    }
  }
/>
`;

exports[`ObjectField renders with configured order 1`] = `
<React.Fragment>
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={true}
    locale="pt-br"
    path="user.email"
    schema={
      Object {
        "type": "string",
      }
    }
  />
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="user.firstName"
    schema={
      Object {
        "type": "string",
      }
    }
  />
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="user.lastName"
    schema={
      Object {
        "type": "string",
      }
    }
  />
</React.Fragment>
`;

exports[`ObjectField renders with empty order 1`] = `<React.Fragment />`;

exports[`ObjectField renders with required 1`] = `
<React.Fragment>
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={true}
    locale="pt-br"
    path="user.email"
    schema={
      Object {
        "type": "string",
      }
    }
  />
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="user.firstName"
    schema={
      Object {
        "type": "string",
      }
    }
  />
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="user.lastName"
    schema={
      Object {
        "type": "string",
      }
    }
  />
</React.Fragment>
`;

exports[`ObjectField renders with title and required 1`] = `
<ObjectFieldWrapper
  config={Object {}}
  formName="test-form"
  isRequired={true}
  locale="pt-br"
  path="user"
  schema={
    Object {
      "properties": Object {
        "email": Object {
          "type": "string",
        },
        "firstName": Object {
          "type": "string",
        },
        "lastName": Object {
          "type": "string",
        },
      },
      "title": "User",
      "type": "object",
    }
  }
>
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="user.firstName"
    schema={
      Object {
        "type": "string",
      }
    }
  />
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="user.lastName"
    schema={
      Object {
        "type": "string",
      }
    }
  />
  <SchemaField
    config={Object {}}
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="user.email"
    schema={
      Object {
        "type": "string",
      }
    }
  />
</ObjectFieldWrapper>
`;

exports[`ObjectField renders without required 1`] = `
<React.Fragment>
  <SchemaField
    config={Object {}}
    context={
      Object {
        "foo": "bar",
      }
    }
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="email"
    schema={
      Object {
        "type": "string",
      }
    }
    widgets={Object {}}
  />
  <SchemaField
    config={Object {}}
    context={
      Object {
        "foo": "bar",
      }
    }
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="firstName"
    schema={
      Object {
        "type": "string",
      }
    }
    widgets={Object {}}
  />
  <SchemaField
    config={Object {}}
    context={
      Object {
        "foo": "bar",
      }
    }
    formName="test-form"
    isRequired={false}
    locale="pt-br"
    path="lastName"
    schema={
      Object {
        "type": "string",
      }
    }
    widgets={Object {}}
  />
</React.Fragment>
`;
